onnxruntime/contrib_ops/cuda/layer_norm_impl.cu

```diff
@@ -326,18 +326,18 @@ __global__ void cuApplyLayerNorm(
     U mu, sigma2;
     cuWelfordMuSigma2<T, U, simplified>(vals, n1, n2, i1, mu, sigma2, buf);
     const T* lvals = vals + i1 * n2;
-    T* ovals = output_vals + i1 * n2;
+    V* ovals = output_vals + i1 * n2;
     U c_inv_std_dev = rsqrt(sigma2 + epsilon);
```

Comment:
Seems 'U' is used to do the safe operation. it is specified as attribute of LayerNorm op, named 'stash_type'.

Reply:
So U type is used for calculation, then cast to V finally into ovals[i].